### PR TITLE
[AAE-12288] Enable Environment Id filtering support

### PR DESCRIPTION
**Pull request by DiogoABastos**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Process instances, user tasks and service tasks ADF components did not support environment id filtering.

**What is the new behaviour?**
Process instances, user tasks and service tasks ADF components now support environment id filtering.

**Does this PR introduce a breaking change?** (check one with "x")

> - [x] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts [hunk 1]
```diff
@@ -76,6 +79,9 @@ export class EditProcessFilterCloudComponent implements OnInit, OnChanges, OnDes
     @Input()
     actions = DEFAULT_ACTIONS;
 
+    @Input()
+    environmentList: any[] = [];
```
Comment (alep85) [inline lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts:83]: we should avoid the use of `any` type if is possible

File: lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts [hunk 2]
```diff
@@ -399,12 +409,21 @@ export class EditProcessFilterCloudComponent implements OnInit, OnChanges, OnDes
             .subscribe((applications) => {
                 if (applications && applications.length > 0) {
                     applications.map((application) => {
-                        this.applicationNames.push({ label: application.name, value: application.name });
+                        if (application.environmentId) {
```
Comment (alep85) [inline lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts:412]: unit tests are missing

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 3]
```diff
@@ -67,6 +67,12 @@ export abstract class BaseEditTaskFilterCloudComponent<T> implements OnInit, OnC
     @Input()
     id: string;
 
+    @Input()
+    environmentId: string;
+
+    @Input()
+    environmentList: any[] = [];
```
Comment (alep85) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:74]: same here about avoiding `any`

**Review by alep85 — CHANGES_REQUESTED** [review]

File: lib/process-services-cloud/src/lib/common/public-api.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export * from './interface/index';
+export * from './mock/index';
```
Comment (alep85) [inline lib/process-services-cloud/src/lib/common/public-api.ts:19]: I'm not sure we need to export the mock, we shouldn't add it to the final bundle
@eromano @popovicsandras what do you think?

**Review by eromano — CHANGES_REQUESTED** [review]
can you update also the documentation?

**Review by andrea-ligios — APPROVED** [review]

**Review by eromano — APPROVED** [review]

**Pull request merged**